// express/tests/karma.conf.js
module.exports = function (config) {
    config.set({

        basePath: '../',

        files: [
            'tests/unit/**/*.js'
        ],

        preprocessors: {
          'tests/unit/**/*.js': [ 'browserify' ]
        },

        browserify: {
          debug: true,
          transform: [ 'brfs' ],
          configure: function(bundle) {
            bundle.on('prebundle', function() {
              bundle.require('./public/node_modules/angular', {expose: 'angular'});
              bundle.require('./public/node_modules/angular-cookies', {expose: 'angular-cookies'});
              bundle.require('./public/node_modules/angular-mocks', {expose: 'angular-mocks'});
              bundle.require('./public/node_modules/angular-socket.io-mock/', {expose: 'angular-socket.io-mock'});
            });
          }
        },

        autoWatch: true,

        singleRun: false,

        frameworks: ['jasmine', 'browserify'],

        browsers: ['Chrome', 'PhantomJS'],

        plugins: [
            'karma-chrome-launcher',
            'karma-jasmine',
            'karma-phantomjs-launcher',
            'karma-browserify'
        ],

        junitReporter: {
            outputFile: 'test_out/unit.xml',
            suite: 'unit'
        }

    });
};
